File: SDKs/googleapis.com/datastream/v1alpha1/python/src/sdk/models/shared/destinationconfig.py
import dataclasses
from typing import Optional
from dataclasses_json import dataclass_json
from sdk import utils
from ..shared import gcsdestinationconfig as shared_gcsdestinationconfig


@dataclass_json
@dataclasses.dataclass
class DestinationConfig:
    r"""DestinationConfig
    The configuration of the stream destination.
    """
    
    destination_connection_profile_name: Optional[str] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('destinationConnectionProfileName') }})
    gcs_destination_config: Optional[shared_gcsdestinationconfig.GcsDestinationConfig] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('gcsDestinationConfig') }})
    

File: SDKs/googleapis.com/datastream/v1alpha1/java/lib/src/main/java/openapisdk/models/shared/DestinationConfig.java
package openapisdk.models.shared;

import com.fasterxml.jackson.annotation.JsonProperty;
import com.fasterxml.jackson.annotation.JsonInclude;
import com.fasterxml.jackson.annotation.JsonInclude.Include;

/**
 * DestinationConfig
 * The configuration of the stream destination.
**/
public class DestinationConfig {
    @JsonInclude(Include.NON_ABSENT)
    @JsonProperty("destinationConnectionProfileName")
    public String destinationConnectionProfileName;
    public DestinationConfig withDestinationConnectionProfileName(String destinationConnectionProfileName) {
        this.destinationConnectionProfileName = destinationConnectionProfileName;
        return this;
    }
    @JsonInclude(Include.NON_ABSENT)
    @JsonProperty("gcsDestinationConfig")
    public GcsDestinationConfig gcsDestinationConfig;
    public DestinationConfig withGcsDestinationConfig(GcsDestinationConfig gcsDestinationConfig) {
        this.gcsDestinationConfig = gcsDestinationConfig;
        return this;
    }
}